doc/xplat_cli_migration.md

```diff
@@ -0,0 +1,223 @@
+# Migrating from Azure XPlat CLI to Azure CLI 2.0
+
+With the introduction of our Azure CLI 2.0 preview, customers now have a
+choice of Azure command-line tools.  While our Azure CLI 2.0 is built to be
+easier to use and automate, please consider the following before starting:
+
+* Both Azure CLI's can be installed and used side-by-side
+* The Azure CLI 2.0 will not support ASM/Classic mode services
+* Scripts are not compatible between both CLIs
+
+## Why consider trying the Azure CLI 2.0?
+Regardless of the tool or vendor, it is important to ensure you benefit when
+adding a new tool to your toolbox.  While the following covers only a small
+number of difference between these two products, we expect customers to benefit
+from the following:
+
+* Clean outputs for common workflows
+  * `--out table` for simplified human output
+  * `--out json` and `--out jsonc` for JSON outputs
+  * `--out tsv` for interacting with tools like grep, AWK, and jq
+* Improved and consistent in-tool documentation with `-h`
+  * Includes descriptions for groups, commands, and parameters
+  * Indicates where to look up required values
+  * Examples and links to web content
+* Improved command-line productivity
+  * Use `[tab][tab]` to lookup parameters, including resource groups and names (only supported in BASH and BASH on Windows)
+  * Work with either ID values (`--ids`) _or_ resource group and name (`-g -n`)
+  * Built in client-side query engine powered by JMESPath
+* (Coming) Support for more features and services
+  * Our preview meets or exceeds XPlat CLI functionality for Compute, Storage, Network, RBAC, and ARM
+  * More services coming online soon!
+
+While we believe the above list is compelling, it's important to remember
+**the Azure CLI 2.0 only supports ARM mode**.  You must continue to use
+the Azure XPlat CLI for ASM/Classic mode management.
+
+## Getting both CLI's setup side-by-side
+
+First, you run `azure --version` and ensure you are using `0.10.5` or later, as
+this is required for sharing your credentials between both CLIs.  If you installed
+using NPM, upgrade with `npm upgrade -g azure-cli`.  If you used an installer,
+we recommend downloading the latest installer to upgrade.
```

Comment:
Please note that the latest azure-xplat-cli may not work against azure stack and I don't know if it works against all global clouds.